### PR TITLE
chore: try adding codecov

### PR DESCRIPTION
**Pull request by JamesHenry**
I haven't done this before, so will likely mess it up. Initially copied config file from prettier

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,7 +18,9 @@ script:
   - yarn check-format
   - yarn test
   - yarn build
+  - yarn global add codecov
 after_success:
+  - yarn codecov -f coverage/cobertura-coverage.xml
```
Comment (j-f1) [inline .travis.yml]: You shouldn’t need the `yarn` here since `codecov` will be globally installed.

File: package.json [hunk 2]
```diff
@@ -62,7 +62,8 @@
     "commitmsg": "commitlint -E GIT_PARAMS",
     "check-format": "prettier --list-different \"./**/*.{ts,js,json,md}\"",
     "semantic-release": "semantic-release",
-    "travis-deploy-once": "travis-deploy-once"
+    "travis-deploy-once": "travis-deploy-once",
+    "codecov": "codecov"
```
Comment (j-f1) [inline package.json]: Same here.

File: .travis.yml [hunk 3]
```diff
@@ -18,7 +18,9 @@ script:
...
   - npm run travis-deploy-once "npm run semantic-release"
```
Comment (armano2) [inline .travis.yml]: why we call here npm instead of yarn?

Comment (JamesHenry) [reply .travis.yml]: Just because of the history of the project, not a big deal either way but I will update it

**Comment by j-f1** [comment]
Looks like you’ve got an error:

```
X Failed to read file at /home/travis/build/JamesHenry/typescript-estree/coverage/cobertura-coverage.xml
```

**Comment by JamesHenry** [comment]
Thanks so much @j-f1 and @armano2 - this is laziest dev work I've ever done, not even bothering to look at the docs of the product I'm adding and let you two find all the issues for me 😂

**Comment by armano2** [comment]
looks like its working https://codecov.io/gh/JamesHenry/typescript-estree/pull/114/tree?path=src

**Comment by JamesHenry** [comment]
Yep I'll just update the coverage level to be appropriate with the starting point

**Comment by armano2** [comment]
i have few question about this, can we turn off coverage on dev builds, and enable jest ci on ci builds?
```
yarn test -- --coverage --ci
```

and in jest.config.js
```
collectCoverage: false
```

jest has bug that you can't debug when `collectCoverage` is set to true. 

---------

`--ci`:
> When this option is provided, Jest will assume it is running in a CI environment. This changes the behavior when a new snapshot is encountered. Instead of the regular behavior of storing a new snapshot automatically, it will fail the test

`--coverage`:
> Indicates that test coverage information should be collected and reported in the output.

**Comment by JamesHenry** [comment]
For sure! Just wanted to make sure I got the infrastructure enabled ASAP -
I’m definitely open to any reasonable config changes

**Pull request merged**